extern-api-java-sdk-common/src/main/java/ru/kontur/extern_api/sdk/model/WrappedCheckResultDataTaskInfo.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2019 SKB Kontur
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package ru.kontur.extern_api.sdk.model;
+
+public class WrappedCheckResultDataTaskInfo extends TaskInfo<DataWrapper<CheckResultData>> {
+
+    /**
+     * Unwrap wrapper internal data
+     *
+     * @return @see CheckResultDataTaskInfo
+     */
+    public CheckResultDataTaskInfo unwrap() {
```

Comment:
Можно написать два помогающих метода: 
```java
class TaskInfo<TResult> {
    ...
   
    public <TOut> TaskInfo<TOut> map(Function<TResult, TOut> mapper) {
        return map(TaskInfo::new, mapper);
    }

    public <TOut, TaskOut extends TaskInfo<TOut>> TaskOut map(
            Supplier<TaskOut> constructor,
            Function<TResult, TOut> mapper
    ) {
        TaskOut task = constructor.get();
        TaskInfo<TOut> taskOut = task;
        taskOut.error = error;
        taskOut.id = id;
        taskOut.taskState = taskState;
        taskOut.taskType = taskType;
        if (isSuccessful() && taskResult != null) {
            taskOut.taskResult = mapper.apply(taskResult);
        }
        return task;
    }
}

```

И тогда
```suggestion
    public CheckResultDataTaskInfo unwrap() {
        return map(CheckTaskInfo::new, DataWrapper::getData);
    }
```
И ещё и переиспользуемо

Reply:
done